lib/core/time.toit

```diff
@@ -726,32 +739,35 @@ class Time implements Comparable:
         zone_is_adjusted = true
         cut := plus > 0 ? plus : minus
         zone_parts := str[cut + 1..]
-        zone_minutes = 60 * (str_to_int.call zone_parts[..2])
-        if zone_parts.size == 4:  // +HHMM
-          zone_minutes += str_to_int.call zone_parts[2..]
-        else if zone_parts.size == 5:  // +HH:MM
-          if (zone_parts[2] != ':'): throw "INVALID_ARGUMENT"
-          zone_minutes += str_to_int.call zone_parts[3..]
-        else if zone_parts.size != 2:  // +HH
-          throw "INVALID_ARGUMENT"
+        // RFC 3339 requires the zone to be in the form hh:mm.
+        if zone_parts.size != 5: throw "INVALID_ARGUMENT"
+        zone_parts.split ":":
+          if it.size != 2: throw "INVALID_ARGUMENT"
+          zone_minutes *= 60
+          zone_minutes += str_to_int.call it
         zone_minutes = plus > 0 ? -zone_minutes : zone_minutes
         str = str[..cut]
     parts ::= str.split "T"
     if parts.size != 2: throw "INVALID_ARGUMENT"
     date_parts ::= (parts[0].split "-").map str_to_int
     if date_parts.size != 3: throw "INVALID_ARGUMENT"
-    time_parts ::= (parts[1].split ":").map str_to_int
+    time_parts ::= parts[1].split ":"
     if time_parts.size != 3: throw "INVALID_ARGUMENT"
+    h := str_to_int.call time_parts[0]
+    m := str_to_int.call time_parts[1]
+    s_float := float.parse time_parts[2]  // Float.parse has no --on_error parameter.
```

Comment:
Wouldn't it be better to split the seconds and then int-parse the nano-seconds?

This is how I did it in my version:
```
    if time_string_parts[2].contains ".":
      split := time_string_parts[2].split "."
      time_string_parts[2] = split[0]
      time_string_parts.add "$split[1]000000000"[..9]
    else:
      time_string_parts.add "0"
    time_parts := time_string_parts.map: int.parse it
    return Time.local_or_utc_
      date_parts[0]
      date_parts[1]
      date_parts[2]
      time_parts[0]
      time_parts[1]
      time_parts[2]
      --ms=0
      --us=0
      --ns=time_parts[3]
      --is_utc=is_utc
```